std/range/package.d

```diff
@@ -3370,21 +3370,24 @@ struct Cycle(R)
         /// ditto
         enum opDollar = DollarToken.init;
 
-        /// ditto
-        auto opSlice(size_t i, size_t j)
-        in
-        {
-            assert(i <= j);
-        }
-        body
+        static if (hasSlicing!R)
         {
-            return this[i .. $].takeExactly(j - i);
```

Comment:
@9il @andralex Doesn't matter, infinite ranges are already excluded from this path on line 3289